src/components/smartSearch/filters/MostActive.tsx

```diff
@@ -0,0 +1,93 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useState } from 'react';
+
+import { isFilterWithId } from '../utils';
+import StyledNumberInput from '../inputs/StyledNumberInput';
+import StyledSelect from '../inputs/StyledSelect';
+import TimeFrame from './TimeFrame';
+import { ZetkinSmartSearchFilter } from 'types/zetkin';
+import { ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+
+interface MostActiveProps {
+    filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter | null;
+    onSubmit: (filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter) => void;
+    onCancel: () => void;
+}
+
+const MostActive = ({ onSubmit, onCancel, filter }: MostActiveProps): JSX.Element => {
+    const [numPeople, setNumPeople] = useState(filter?.config?.size || 10);
+    const [timeFrame, setTimeFrame] = useState({});
+    const [op, setOp] = useState<'add' | 'sub'>(filter?.op || 'add');
+
+    const handleAddFilter = (e: FormEvent) => {
+        e.preventDefault();
+        onSubmit({
+            config: {
+                size: numPeople,
+                ...timeFrame,
+            },
+            ...(filter && isFilterWithId(filter) && { id: filter.id }),
+            op,
+            type: 'most_active',
+        });
+    };
+
+    const handleTimeFrameChange = (range: {after?: string; before?: string}) => {
+        setTimeFrame(range);
+    };
+
+    return (
+        <form onSubmit={ e => handleAddFilter(e) }>
+            <Typography style={{ lineHeight: 'unset', marginBottom: '2rem' }} variant="h4">
+                <Msg id="misc.smartSearch.most_active.inputString" values={{
+                    addRemoveSelect: (
+                        <StyledSelect onChange={ e => setOp(e.target.value as 'add' | 'sub') }
+                            value={ op }>
+                            <MenuItem key="add" value="add">
+                                <Msg id="misc.smartSearch.most_active.addRemoveSelect.add"/>
+                            </MenuItem>
+                            <MenuItem key="sub" value="sub">
+                                <Msg id="misc.smartSearch.most_active.addRemoveSelect.sub" />
+                            </MenuItem>
+                        </StyledSelect>
+                    ),
+                    numPeople: numPeople,
+                    numPeopleSelect: (
+                        <StyledNumberInput
+                            defaultValue={ numPeople }
+                            onChange={ (e) => setNumPeople(+e.target.value) }
+                        />
+                    ),
+                    timeFrame: (
+                        <TimeFrame filterConfig={{
+                            ...(filter?.config?.after && { after: filter.config.after }),
```

Comment:
Is this any better than `after: filter?.config?.after` (which will default to `undefined`)?

Reply:
this should have been resolved in #310